zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -261,6 +261,12 @@ public int getNumIdleClient() {
     }
   }
 
+  public void setMaxPoolSize(int size) {
+    if (clientPool != null) {
+      //Size + 2 for progress poller , cancel operation
+      clientPool.setMaxTotal(size + 2);
```

Comment:
The Link is pointing to `GenericKeyedObjectPoolConfig`.
The source for `GenericObjectPoolConfig` is [here](https://commons.apache.org/proper/commons-pool/apidocs/src-html/org/apache/commons/pool2/impl/GenericObjectPoolConfig.html)


In here, the  `DEFAULT_MAX_TOTAL = 8; `